getsentry/sentry-javascript-bundler-plugins
# test(e2e): Add E2E tests

Alright, here we go...

This PR adds basic E2E or Snapshot tests to our repo in a dedicated package. The high-level idea is to add a set of input JS files, a plugin config and a set of reference files per test scenario. For a more detailed explanation, take a look at the README file. In addition, it adds one test scenario which tests a very basic sourcemaps upload configuration.

Here's what these tests do:

Per test scenario, per bundler:
1. Delete the release from Sentry.io (sentry-sdks org) if it is already present
2. Bundle the input files and run the Sentry Bundler Plugin which uploads the files to Sentry
3. Download the release from Sentry 
4. Compare downloaded files against reference files. They must match exactly for the test to pass. 

As of right now, it is arguable if these tests should have their own package or be in the integration-tests package. I opted for the former because there are subtle differences and if we chose to improve them (e.g. do the Verdaccio approach from the JS SDK or add recipes) the structure and components will differ more. 

Admittedly, there's definitely room for improvement for these tests but I've already been spending too much time on them and we can always iterate here to end up with something better down the line.

FAQ:

Q: Why do we have to delete the release first?
A: Every new release we make changes generated JS and sourcemaps because of our plugin's release injection functionality. This means that our ref files would no longer match, letting our tests fail. Therefore, we delete the old release and recreate the very same release again. Also, this sort of helps keeping an overview over different testing scenarios and their releases in the Sentry.io project.

Q: CI?
A: Yes, coming soon (TM)

fixes #70
